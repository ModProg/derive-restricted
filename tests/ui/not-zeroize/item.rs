use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(crate = struct Test)]
struct InvalidPath<T>(PhantomData<T>);

#[derive_where(skip_inner, Clone)]
struct SkipInnerWithTrait<T>(PhantomData<T>);

#[derive_where(,)]
struct OnlyComma<T>(PhantomData<T>);

#[derive_where(,Clone)]
struct StartWithComma<T>(PhantomData<T>);

#[derive_where(Clone,,)]
struct DuplicateCommaAtEnd<T>(PhantomData<T>);

#[derive_where("Clone")]
struct InvalidTrait<T>(PhantomData<T>);

#[derive_where(T)]
struct GenericInsteadTrait<T, U>(T, PhantomData<U>);

fn main() {}
