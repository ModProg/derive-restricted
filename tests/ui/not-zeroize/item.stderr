error: unsupported trait, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
 --> tests/ui/not-zeroize/item.rs:5:16
  |
5 | #[derive_where(skip_inner, Clone)]
  |                ^^^^^^^^^^

error: unsupported trait syntax, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
 --> tests/ui/not-zeroize/item.rs:8:16
  |
8 | #[derive_where(,)]
  |                ^

error: unsupported trait syntax, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:11:16
   |
11 | #[derive_where(,Clone)]
   |                ^

error: unsupported trait syntax, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:14:22
   |
14 | #[derive_where(Clone,,)]
   |                      ^

error: unsupported trait syntax, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:17:16
   |
17 | #[derive_where("Clone")]
   |                ^^^^^^^

error: unsupported trait, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:20:16
   |
20 | #[derive_where(T)]
   |                ^
